src/main/java/com/meilisearch/sdk/Documents.java

```diff
@@ -21,82 +22,162 @@ protected Documents(Config config) {
     /**
      * Retrieves the document at the specified index uid with the specified identifier
      *
+     * @param <T> Type of the document returned
      * @param uid Partial index identifier for the requested documents
      * @param identifier ID of the document
-     * @return String containing the requested document
+     * @param targetClass Class of the document returned
+     * @return Object containing the requested document
+     * @throws MeilisearchException if the client request causes an error
+     */
+    <T> T getDocument(String uid, String identifier, Class<T> targetClass)
```

Comment:
You have to also add support to `fields`, which allows the user to reduce the payload meilisearch will return like we have here: https://github.com/meilisearch/meilisearch-python/blob/a6dc0631103a495d109ff6b69e288193d1b22b73/tests/index/test_index_document_meilisearch.py#L71-L78.

Reply:
This can be used in the `getDocuments`, `getDocument`, `getRawDocuments`, and `getRawDocument`